st2client/st2client/commands/auth.py

```diff
@@ -70,6 +73,84 @@ def run_and_print(self, args, **kwargs):
                               attributes=self.display_attributes, json=args.json, yaml=args.yaml)
 
 
+class LoginCommand(resource.ResourceCommand):
+    display_attributes = ['user', 'token', 'expiry']
+
+    def __init__(self, resource, *args, **kwargs):
+
+        kwargs['has_token_opt'] = False
+
+        super(LoginCommand, self).__init__(
+            resource, kwargs.pop('name', 'create'),
+            'Authenticate user, acquire access token, and update CLI config directory',
+            *args, **kwargs)
+
+        self.parser.add_argument('username',
+                                 help='Name of the user to authenticate.')
+
+        self.parser.add_argument('-p', '--password', dest='password',
+                                 help='Password for the user. If password is not provided, '
+                                      'it will be prompted.')
+        self.parser.add_argument('-l', '--ttl', type=int, dest='ttl', default=None,
+                                 help='The life span of the token in seconds. '
+                                      'Max TTL configured by the admin supersedes this.')
+        self.parser.add_argument('-w', '--write-real-password', action='store_true', default=False,
+                                 dest='write_password',
+                                 help='Write the real (plain-text) password to the config file '
+                                      '(default is to omit it')
+
+    def run(self, args, **kwargs):
+
+        if not args.password:
+            args.password = getpass.getpass()
+        instance = self.resource(ttl=args.ttl) if args.ttl else self.resource()
+
+        cli = BaseCLIApp()
+
+        # Determine path to config file
+        try:
+            config_file = cli._get_config_file_path(args)
+        except ValueError:
+            # config file not found in args or in env, defaulting
+            config_file = expanduser('~/.st2/config')
```

Comment:
Can you use os.path.expanduser instead and import os above?  Also, can you use the constants at https://github.com/StackStorm/st2/blob/master/st2client/st2client/config_parser.py#L36 instead of duplicating and hard coding the path value here?